Apps/Inc/common.h

```diff
@@ -28,6 +28,19 @@
 #define EXPOSE_GETTER(type, name) \
     type get_##name(void); \
 
+/**
+ * Used for creating setter functions (sets the value based on given inputs) 
+ * 
+ * SETTER is used for creating the function and EXPOSE_SETTER is used for creating the declaration in the header file
+ */
+#define SETTER(type, name) \
+    void set_##name(type val){ \
+        name = val; \
+    } \
+
+#define EXPOSE_SETTER(type, name) \
+    void set_##name(type val); \
+
```

Comment:
Could you define mock & stub macro?

Reply:
Also, I don't know if it makes much sense to macro out setters because you might want to use setters in general. We don't do so now but in general, maybe in the future, a module could us a setter as its normal functionality. But if you're adamant on macro-ing them out, that's fine